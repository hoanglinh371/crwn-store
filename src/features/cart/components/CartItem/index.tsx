import React from 'react';
import classNames from 'classnames/bind';

import { CartItem as CartItemProps } from 'src/interfaces/product.interface';
import styles from './CartItem.module.scss';

interface Props {
  cartItem: CartItemProps;
}

const cx = classNames.bind(styles);

const CartItem: React.FC<Props> = ({ cartItem }) => {
  const { name, imageUrl, quantity, price } = cartItem;

  return (
    <div className={cx('container')}>
      <img src={imageUrl} alt={name} />
      <div className={cx('item-details')}>
        <span className={cx('name')}>{name}</span>
        <span className={cx('price')}>
          {quantity} x ${price}
        </span>
      </div>
    </div>
  );
};

export default CartItem;
