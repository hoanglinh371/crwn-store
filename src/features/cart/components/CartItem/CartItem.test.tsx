import { cleanup, render, screen } from '@testing-library/react';

import { CartItem as CartItemProps } from 'src/interfaces/product.interface';
import CartItem from '.';

const mockCartItem: CartItemProps = {
  id: 1,
  name: 'Koi Ring',
  imageUrl: 'koi-ring.png',
  price: 50,
  quantity: 2,
};

describe('CartItem component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(<CartItem cartItem={mockCartItem} />);
    expect(container).toMatchSnapshot();
  });

  it('Should_HaveImageSrc', () => {
    render(<CartItem cartItem={mockCartItem} />);

    expect(screen.getByLabelText('image')).toHaveAttribute(
      'src',
      'koi-ring.png',
    );
  });
});
