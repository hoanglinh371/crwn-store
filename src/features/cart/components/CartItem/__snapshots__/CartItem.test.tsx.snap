// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="Koi Ring"
      aria-label="image"
      src="koi-ring.png"
    />
    <div
      class="item-details"
    >
      <span
        class="name"
      >
        Koi Ring
      </span>
      <span
        class="price"
      >
        2
         x $
        50
      </span>
    </div>
  </div>
</div>
`;
